google/resource_storage_bucket.go

```diff
@@ -1119,13 +1129,19 @@ func expandBucketWebsite(v interface{}) *storage.BucketWebsite {
 }
 
 func expandIamConfiguration(d *schema.ResourceData) *storage.BucketIamConfiguration {
-	return &storage.BucketIamConfiguration{
+	cfg := &storage.BucketIamConfiguration{
 		ForceSendFields: []string{"UniformBucketLevelAccess"},
 		UniformBucketLevelAccess: &storage.BucketIamConfigurationUniformBucketLevelAccess{
 			Enabled:         d.Get("uniform_bucket_level_access").(bool),
 			ForceSendFields: []string{"Enabled"},
 		},
 	}
```

Comment:
This is changing existing behavior... we should craft the object manually so that we only configure the object with these properties  if `UniformBucketLevelAccess` is present on the config. I recommend checking `uniform_bucket_level_access` and configuring it appropriately.

Reply:
Isn't that what the code currently does at L592? `expandIamConfiguration` is only called if:

```golang
	if d.HasChange("uniform_bucket_level_access") || d.HasChange("public_access_prevention") {
		sb.IamConfiguration = expandIamConfiguration(d)
	}
```

This check was already there, I added the test for `public_access_prevention`.

The tests look promising locally, although the CI/CD workflow looks like it still needs approval. I'll prepare the upstream changes now.

Reply:
Raised https://github.com/GoogleCloudPlatform/magic-modules/pull/5519 and tagged you there. Thanks!